docs/src/developers_guide/release.rst

```diff
@@ -226,7 +226,6 @@ Release Steps
 
 #. Create the release feature branch ``v1.9.x`` on `SciTools/iris`_.
```

Comment:
We didn't do it this way round for `v3.3` - creating the branch came after the What's New changes. Which do you reckon is correct? We struggled to agree at the time.